src/txn/txn_rollback_to_stable.c

```diff
@@ -173,7 +176,17 @@ __rollback_abort_update(WT_SESSION_IMPL *session, WT_ITEM *key, WT_UPDATE *first
         }
         if (stable_update_found != NULL)
             *stable_update_found = true;
-    }
+    } else
+      /*
+       * When stable update cleanup isn't performed make sure any unstable updates in the history
+       * store are removed. An unstable update in the data store will be removed when the page is
+       * next reconciled.
+       *
+       * FIXME-WT-10017: WT-9846 is an interim fix while we investigate the impacts of a longer term
+       * correction in WT-10017. Once completed this change can be reverted.
+       */
+      if (oldest_unstable_hs_upd != NULL)
+        WT_RET(__rollback_delete_hs(session, key, oldest_unstable_hs_upd->start_ts));
```

Comment:
Why not just pass the stable timestamp here?

Reply:
I think we want to delete anything that is newer than the stable timestamp in hs.

Reply:
No. That can go wrong. For example, if the on-disk update is also unstable, the above call removes all the required updates can lead to wrong data being restored into the data store.

Reply:
ok